msticpy/data/core/query_provider_connections_mixin.py

```diff
@@ -81,18 +101,361 @@ def list_connections(self) -> List[str]:
+
+        Other Parameters
+        ----------------
+        debug: bool, optional
+            Return queries to be executed rather than execute them, by default False
+        progress: bool, optional
+            Show progress bar, by default True
+        retry: bool, optional
+            Retry failed queries, by default False
+        **kwargs : Dict[str, Any]
+            Additional keyword arguments to pass to the query method.
+
+        Returns
+        -------
+        pd.DataFrame
+            The concatenated results of the query executed against all connections.
+
+        Notes
+        -----
+        This method executes the time-chunks of the split query.
+        If the driver supports threading or async execution, the sub-queries are
+        executed asynchronously. Otherwise, the queries are executed sequentially.
+
+        """
+        start = query_params.pop("start", None)
+        end = query_params.pop("end", None)
+        progress = kwargs.pop("progress", True)
+        retry = kwargs.pop("retry_on_error", False)
+        debug = kwargs.pop("debug", False)
+        if not (start or end):
+            print(
+                "Cannot split a query that does not have 'start' and 'end' parameters"
+            )
+            return None
+        try:
+            split_delta = pd.Timedelta(split_by)
+        except ValueError:
+            split_delta = pd.Timedelta("1D")
+        logger.info("Using split delta %s", split_delta)
+
+        ranges = _calc_split_ranges(start, end, split_delta)
+
+        split_queries = {
+            (q_start, q_end): query_source.create_query(
+                formatters=self._query_provider.formatters,
+                start=q_start,
+                end=q_end,
+                **query_params,
+            )
+            for q_start, q_end in ranges
+        }
+        logger.info("Split query into %s chunks", len(split_queries))
+        if debug:
+            return "\n\n".join(
+                f"{start}-{end}\n{query}"
+                for (start, end), query in split_queries.items()
+            )
+
+        # Retrieve any query options passed (other than query params)
+        # and send to query function.
+        query_options = self._get_query_options(query_params, kwargs)
+        logger.info("query_options: %s", query_options)
+        logger.info("kwargs: %s", kwargs)
+        if "time_span" in query_options:
+            del query_options["time_span"]
+        query_tasks = {
+            f"{start}-{end}": partial(
+                self.exec_query,
+                query=query_str,
+                query_source=query_source,
+                time_span={"start": start, "end": end},
+                **query_options,
+            )
+            for (start, end), query_str in split_queries.items()
+        }
+        # Run the queries threaded if supported
+        if self._query_provider.get_driver_property(DriverProps.SUPPORTS_THREADING):
```

Comment:
This block seems repetitious with https://github.com/microsoft/msticpy/pull/645/files#diff-8c03afafd2896629d2c9a856c9416072a595ba61621d02f3e5427cf267b692afR156. Might be an easy refactor into a separate function.